From a12473942ea0c9ea8ae531f80fdb94f20730702b Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Tue, 14 Sep 2010 13:58:01 +0800
Subject: [PATCH] ENGR00131459 MX53: fix system suspend issue

1. MX53 system suspend is broken by MX50 change
(ENGR00125546). This patch restores suspend iram operation
for MX53.
2. Remove duplicated code in MX50 logic
3. Remove unused variable

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 arch/arm/mach-mx5/pm.c |    9 +++------
 1 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-mx5/pm.c b/arch/arm/mach-mx5/pm.c
index 140b3de..4ce3df5 100644
--- a/arch/arm/mach-mx5/pm.c
+++ b/arch/arm/mach-mx5/pm.c
@@ -79,7 +79,7 @@ static int mx5_suspend_enter(suspend_state_t state)
 		local_flush_tlb_all();
 		flush_cache_all();
 
-		if (cpu_is_mx51()) {
+		if (cpu_is_mx51() || cpu_is_mx53()) {
 			/* Run the suspend code from iRAM. */
 			suspend_in_iram(suspend_param1);
 
@@ -87,9 +87,6 @@ static int mx5_suspend_enter(suspend_state_t state)
 			__raw_writel(0, MXC_SRPG_EMPGC0_SRPGCR);
 			__raw_writel(0, MXC_SRPG_EMPGC1_SRPGCR);
 		} else
-			local_flush_tlb_all();
-			flush_cache_all();
-
 			suspend_in_iram(databahn_base);
 	} else {
 			cpu_do_idle();
@@ -182,7 +179,7 @@ static struct platform_driver mx5_pm_driver = {
 static int __init pm_init(void)
 {
 	int cpu_wp_nr;
-	unsigned long iram_paddr, iram_paddr1;
+	unsigned long iram_paddr;
 
 	pr_info("Static Power Management for Freescale i.MX5\n");
 	if (platform_driver_register(&mx5_pm_driver) != 0) {
@@ -197,7 +194,7 @@ static int __init pm_init(void)
 	suspend_iram_base = __arm_ioremap(iram_paddr, SZ_4K,
 					  MT_HIGH_VECTORS);
 
-	if (cpu_is_mx51()) {
+	if (cpu_is_mx51() || cpu_is_mx53()) {
 		suspend_param1 = IO_ADDRESS(IOMUXC_BASE_ADDR + 0x4b8);
 		memcpy(suspend_iram_base, cpu_do_suspend_workaround,
 				SZ_4K);
-- 
1.5.4.4

